library/manageiq_provider.py

```diff
@@ -234,12 +252,13 @@ def find_provider_by_name(self, provider_name):
         providers = self.client.collections.providers
         return next((p.id for p in providers if p.name == provider_name), None)
 
-    def generate_endpoint(self, role, hostname, port, authtype, token):
+    def generate_endpoint(self, role, authtype, hostname=None, port=None, token=None, userid=None, password=None):
         """ Returns one provider endpoint hash.
         """
         return {'endpoint': {'role': role, 'hostname': hostname,
                              'port': port},
-                'authentication': {'role': authtype, 'auth_key': token}}
+                'authentication': {'authtype': authtype, 'auth_key': token,
+                                   'userid': userid, 'password': password}}
```

Comment:
IMO passing the entire authentication isn't the best user experience but we can work on a solution for the endpoint issue in a separate PR